lib/chef/provider/service/macosx.rb

```diff
@@ -41,11 +44,27 @@ def self.gather_plist_dirs
         PLIST_DIRS = gather_plist_dirs
 
         def load_current_resource
-          @current_resource = Chef::Resource::Service.new(@new_resource.name)
+          @current_resource = Chef::Resource::MacosxService.new(@new_resource.name)
           @current_resource.service_name(@new_resource.service_name)
           @plist_size = 0
-          @plist = find_service_plist
+          @plist = @new_resource.plist ? @new_resource.plist : find_service_plist
+          Chef::Log.debug("#{new_resource} Plist: '#{@plist}'")
```

Comment:
I don't have a strong preference... but I don't think we're gonna block your diff on debug lines. One line is probably nicer this case.